KarhooUISDK/CommonUI/AddressView/AddressView.swift

```diff
@@ -0,0 +1,235 @@
+                )
+            Spacer()
+        }
+    }
+
+    @ViewBuilder
+    private var labelsColumn: some View {
+        VStack(alignment: .leading) {
+            AddressLabelView(
+                text: pickUp.text,
+                subtext: pickUp.subtext
+            )
+            if tags.isNotEmpty {
+                buildTagsView()
+            } else {
+                Spacer(minLength: UIConstants.Spacing.medium)
+            }
+            AddressLabelView(
+                text: destination.text,
+                subtext: destination.subtext
+            )
+        }
+    }
+
+    @ViewBuilder
+    private func buildTimeTextView(_ text: String) -> some View {
+        VStack {
+            Text(text)
+                .font(Font(KarhooUI.fonts.captionBold()))
+                .multilineTextAlignment(.trailing)
+                .padding(.top, UIConstants.Dimension.View.addressViewPadding)
+                .minimumScaleFactor(UIConstants.Dimension.View.addressViewMinimumScaleFactor)
+            Spacer()
+        }
+    }
+
+    @ViewBuilder
+    private func buildTagsView() -> some View {
+        VStack(alignment: .leading) {
+            ForEach(tags) {
+                TagView(tag: $0)
+            }
+        }
+        .padding(.vertical, UIConstants.Spacing.xSmall)
+    }
+}
+
+// MARK: - Address label model & view
+extension KarhooAddressView {
+
+    struct AddressLabel {
+        let text: String
+        let subtext: String
+    }
+
+    struct AddressLabelView: View {
+
+        let text: String
+        var subtext: String?
+
+        var body: some View {
+            VStack(alignment: .leading, spacing: 0) {
+                Text(text)
+                    .font(Font(KarhooUI.fonts.bodyRegular()))
+                    .lineLimit(2)
+                    .multilineTextAlignment(.leading)
+                    .minimumScaleFactor(UIConstants.Dimension.View.addressViewMinimumScaleFactor)
+                    .foregroundColor(Color(KarhooUI.colors.text))
+                    .frame(maxWidth: .infinity, alignment: .leading)
+                if let subtext = subtext {
+                    Text(subtext)
+                        .font(Font(KarhooUI.fonts.captionBold()))
+                        .lineLimit(2)
+                        .multilineTextAlignment(.leading)
+                        .minimumScaleFactor(UIConstants.Dimension.View.addressViewMinimumScaleFactor)
+                        .foregroundColor(Color(KarhooUI.colors.textLabel))
+                        .frame(maxWidth: .infinity, alignment: .leading)
+                }
+            }
+            .frame(height: UIConstants.Dimension.View.addressViewLabelHeight)
+        }
+    }
+}
+
+// MARK: - Tag model & view
+extension KarhooAddressView {
+    struct Tag: Identifiable {
+        let id = UUID()
+        let title: String
+    }
+
+    struct TagView: View {
+
+        let tag: Tag
+
+        var body: some View {
+            Text(tag.title)
```

Comment:
Can we use any pre-defined UIConstants for the height, padding, etc?

Reply:
Unfortunetly no, if we want to make the feature aligned with the design. It's very case-specific.